rasa cli: use default config

tabergma opened the pull request:
**Proposed changes**:
- `rasa train` does not fail anymore on invalid or not-existing config
- `rasa train` uses a default config if the given config is not valid and prints a user warning

related to #3272 

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

erohmensing commented:
I like this so far, but can we make it choose only the undefined properties from the default config? Right now, if I don't define policies, then my defined nlu config gets overwritten, and vice versa, which isn't ideal

tabergma commented:
I updated the code:
If a config contains just a few keys, missing keys are added using the fallback configuration values.
Also fixed the issue with a non-existing config.
